apps/factory/src/components/marketplace/like_button.tsx

```diff
@@ -1,14 +1,76 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import Button from "@mui/material/Button";
 import ButtonGroup from "@mui/material/ButtonGroup";
 import FavoriteIcon from "@mui/icons-material/Favorite";
 
-function LikeButton({ count }: { count: number }) {
-  const [counter, setCounter] = useState(count);
+import { api } from "~/utils/api";
+import { signIn, useSession } from "next-auth/react";
 
-  const handleLikeClick = () => {
-    // Increment the like count when the button is clicked
-    setCounter(counter + 1);
+interface LikeButtonProps {
+  EntityId: string;
+  EntityType: string;
+}
+
+const LikeButton: React.FC<LikeButtonProps> = ({ EntityId, EntityType }) => {
+  const { data } = api.like.getLikes.useQuery({
+    EntityId,
+    EntityType,
+  });
+
+  const [counter, setCounter] = useState<number>(
+    data?.likesCount != null ? data.likesCount : 0,
+  );
+  const [hasLiked, setHasLiked] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [LikeId, setLikeId] = useState<string>("");
+
+  useEffect(() => {
+    data?.likes.some((like) => {
+      if (like.userId === sessionData?.user.id) {
+        setHasLiked(true);
+        setLikeId(like.likeId!);
+      }
+    });
```

Comment:
@LovedeepSingh19 this is not scalable, you can't pull all the likes to frontend. The user id need to send to the backend and check in db. This need to be changed.

Reply:
You told me to put that code in frontend. Previously in meet I showed you the code and 1st line was checking that same thing